Sources/ParaSwift/Core/ParaManager.swift

```diff
@@ -203,134 +275,641 @@ extension ParaManager {
+        let params = SignUpOrLogInPayload(auth: identity)
+        
+        _ = try await postMessage(method: "loginExternalWallet", payload: params)
+        self.sessionState = .activeLoggedIn
+        
+        logger.debug("External wallet login completed for address: \(wallet.address)")
+    }
+    
+    /// Logs in with an external wallet address (legacy version)
+    /// - Parameters:
+    ///   - externalAddress: The external wallet address
+    ///   - type: The type of wallet (e.g. "EVM")
+    internal func loginExternalWallet(externalAddress: String, type: String) async throws {
+        let walletType = ExternalWalletType(rawValue: type) ?? .evm
+        let wallet = ExternalWalletInfo(address: externalAddress, type: walletType)
+        try await loginExternalWallet(wallet: wallet)
+    }
+    
+    // MARK: - Auth Helper Methods
+    
+    // Private struct for the signUpOrLogIn payload
+    private struct SignUpOrLogInPayload: Encodable {
+        let auth: AnyEncodable
+        
+        init(auth: Encodable) {
+            self.auth = AnyEncodable(auth)
+        }
+    }
+    
+    private func createSignUpOrLogInPayload(from auth: Auth) -> SignUpOrLogInPayload {
+        switch auth {
+        case .email(let emailAddress):
+            return SignUpOrLogInPayload(auth: EmailIdentity(email: emailAddress))
+        case .phone(let phoneNumber):
+            return SignUpOrLogInPayload(auth: PhoneIdentity(phone: phoneNumber, countryCode: phoneNumber.hasPrefix("+") ? "" : ""))
+        case .farcaster(let fid):
+            return SignUpOrLogInPayload(auth: FarcasterIdentity(fid: fid))
+        case .telegram(let id):
+            return SignUpOrLogInPayload(auth: TelegramIdentity(id: id))
+        case .externalWallet(let wallet):
+            return SignUpOrLogInPayload(auth: ExternalWalletIdentity(wallet: wallet))
+        case .identity(let identity):
+            return SignUpOrLogInPayload(auth: identity)
+        }
+    }
+    
+    private func parseAuthStateFromResult(_ result: Any?) throws -> AuthState {
+        guard let resultDict = result as? [String: Any] else {
+            throw ParaError.bridgeError("Invalid result format from authentication call")
+        }
+        
+        guard let stageString = resultDict["stage"] as? String,
+              let stage = AuthStage(rawValue: stageString),
+              let userId = resultDict["userId"] as? String else {
+            throw ParaError.bridgeError("Missing required fields in authentication response")
+        }
+        
+        let passkeyUrl = resultDict["passkeyUrl"] as? String
+        let passkeyId = resultDict["passkeyId"] as? String
+        let passwordUrl = resultDict["passwordUrl"] as? String
+        let passkeyKnownDeviceUrl = resultDict["passkeyKnownDeviceUrl"] as? String
+        let displayName = resultDict["displayName"] as? String
+        let profilePictureUrl = resultDict["profilePictureUrl"] as? String
```

Comment:
Not sure where you're getting `profilePictureUrl` from? It is called `pfpUrl` everywhere I have seen